ansible_pytest_extra_requirements.txt

```diff
@@ -0,0 +1,8 @@
+# SPDX-License-Identifier: MIT
+
+# needs mock and ansible
+mock ; python_version < "3.0"
```

Comment:
Mock is not a dependency for Ansible

Reply:
I guess we need to have 3 files:
pytest_extra_requirements.txt - in the template it is empty - projects can customize to add mock and/or ansible
ansible_pytest_extra_requirements.txt - contains ansible and its dependencies (everything except `mock` from the current ansible_pytest_extra_requirements.txt)
mock_pytest_extra_requirements.txt - contains the `mock` package from the current ansible_pytest_extra_requirements.txt

The comment in `pytest_extra_requirements.txt` will change to say something like "if you use ansible in your pytest, add -ransible_pytest_extra_requirements.txt, and if you use mock, add -rmock_pytest_extra_requirements.txt"

ok?

Reply:
> I guess we need to have 3 files:
> pytest_extra_requirements.txt - in the template it is empty - projects can customize to add mock and/or ansible

Yes, this file would be empty in the template but contain `mock` for the network role.

> ansible_pytest_extra_requirements.txt - contains ansible and its dependencies (everything except `mock` from the current ansible_pytest_extra_requirements.txt)

Sounds good.

> mock_pytest_extra_requirements.txt - contains the `mock` package from the current ansible_pytest_extra_requirements.txt

I do not think we need a file just for mock. Any role that needs mock can just add it to the `pytest_extra_requirements.txt` file.

> The comment in `pytest_extra_requirements.txt` will change to say something like "if you use ansible in your pytest, add -ransible_pytest_extra_requirements.txt, and if you use mock, add -rmock_pytest_extra_requirements.txt, and if you use both, add both -ransible_pytest_extra_requirements.txt and -rmock_pytest_extra_requirements.txt"

This only needs to mention the ansible convenience file.